utils/global-state-update-gen/src/utils/account_manager.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::BTreeMap;
+
+use casper_engine_test_support::LmdbWasmTestBuilder;
+use casper_types::{
+    account::Account, AccessRights, CLValue, Key, PublicKey, StoredValue, URef, U512,
+};
+
+use rand::prelude::*;
+
+use crate::utils::print_entry;
+
+pub struct AccountManager {
+    accounts: BTreeMap<PublicKey, Account>,
+}
+
+impl AccountManager {
+    /// Creates a new instance of `AccountManager`.
+    pub fn new() -> Self {
+        Self {
+            accounts: Default::default(),
+        }
+    }
+
+    /// Gets the account for the given public key.
+    /// If the account doesn't exist, it prints the entries necessary to create it and returns the
+    /// created account.
+    pub fn get_or_create_account(
+        &mut self,
+        builder: &mut LmdbWasmTestBuilder,
+        pkey: &PublicKey,
+    ) -> &Account {
+        // unfortunately, `if let Some(account) = self.accounts.get(&pkey)` upsets the borrow
+        // checker, so we have to do it this way
+        if !self.accounts.contains_key(pkey) {
+            let account_hash = pkey.to_account_hash();
+            if let Some(account) = builder.get_account(account_hash) {
+                let _ = self.accounts.insert(pkey.clone(), account);
+                // unwrapping is safe, we just inserted it
+                self.accounts.get(pkey).unwrap()
+            } else {
+                self.create_account(pkey.clone());
+                // create_account inserts the account, so unwrapping is safe
+                self.accounts.get(pkey).unwrap()
+            }
+        } else {
+            // we know it contains the key, unwrapping safe
+            self.accounts.get(pkey).unwrap()
+        }
+    }
+
+    fn create_account(&mut self, pkey: PublicKey) {
+        let mut rng = rand::thread_rng();
+        let new_purse = URef::new(rng.gen(), AccessRights::READ_ADD_WRITE);
+
+        // Apparently EE does this, not sure why, but we'll also do it for consistency.
+        print_entry(
+            &Key::URef(new_purse),
+            &StoredValue::CLValue(CLValue::unit()),
+        );
+
+        // This will create a new purse by storing a zero balance in it.
+        print_entry(
+            &Key::Balance(new_purse.addr()),
+            &StoredValue::CLValue(CLValue::from_t(U512::zero()).unwrap()),
```

Comment:
Not sure if the account having a zero balance will be a problem.  If so, I guess that's an easy fix.

Reply:
I think it shouldn't be, the mint contract is also able to do that... Although usually for the purpose of transferring some amount of money into it right away. I guess we'll see (or maybe Michał knows better?).

Reply:
> Not sure if the account having a zero balance will be a problem. If so, I guess that's an easy fix.

Account with less than 2.5CSPR will be unusable for deploys.

Reply:
Here they are only used in bids - basically the validator needs to have a purse assigned to their bid. And if I understand correctly, these are the purses needed for unbonding, so it would be as if the validator bid with everything they had... or not?
But now that I think about it, unbonding also requires running a deploy, doesn't it? :grimacing:

Reply:
Yes it does. Wouldn't it be easier to "transfer" some tokens to it?

Reply:
The tokens would have to be transferred from somewhere - so the question would be, who should pay for that?

Reply:
If it's a new validator with a new account that _shouldn't_ be a problem - right now you can bid/delegate 100% of your balance without leaving 2.5CSPR, so creating a new bid entry and associated account with 0 but with valid stake seems OK to me.

But if you need to create a balance, then you also have to modify the total supply.